src/query/pools/query_pools.go

```diff
@@ -116,75 +137,132 @@ func (s sessionPools) TagDecoder() serialize.TagDecoderPool {
 	return s.tagDecoder
 }
 
-func buildBuckets() []pool.Bucket {
-	return []pool.Bucket{
-		{Capacity: defaultBucketCapacity, Count: defaultIdentifierPoolSize},
+// iteratorPoolSize            = 1000
+// checkedBytesWrapperPoolSize = 128
+// defaultIdentifierPoolSize   = 128
+
+// BuildIteratorPoolsOptions is a set of build iterator pools.
+type BuildIteratorPoolsOptions struct {
```

Comment:
nit: maybe better to take in these values as a constructor rather than exposing the internal variables for mutation?